### ISSUE TITLE
vecelement test fails on Zen4

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#53683]
```
keno@demeter1:~/julia$ make test-vecelement 
Whitespace check found no issues.
    JULIA test/vecelement
Running parallel tests with:
  getpid() = 1386043
  nworkers() = 1
  nthreads() = 1
  Sys.CPU_THREADS = 64
  Sys.total_memory() = 755.645 GiB
  Sys.free_memory() = 745.864 GiB

Test   (Worker) | Time (s) | GC (s) | GC % | Alloc (MB) | RSS (MB)
vecelement  (1) |        started at 2024-03-09T23:02:36.957

[1386043] signal 11 (128): Segmentation fault
in expression starting at /home/keno/julia/test/vecelement.jl:113
f20961 at /home/keno/julia/test/vecelement.jl:0
unknown function (ip: 0x7fdd57964b4f)
jl_apply at /home/keno/julia/src/julia.h:2166 [inlined]
do_call at /home/keno/julia/src/interpreter.c:126
eval_value at /home/keno/julia/src/interpreter.c:223
eval_body at /home/keno/julia/src/interpreter.c:560
jl_interpret_toplevel_thunk at /home/keno/julia/src/interpreter.c:815
jl_toplevel_eval_flex at /home/keno/julia/src/toplevel.c:943
jl_toplevel_eval_flex at /home/keno/julia/src/toplevel.c:886
ijl_toplevel_eval_in at /home/keno/julia/src/toplevel.c:994
eval at ./boot.jl:428 [inlined]
include_string at ./loading.jl:2249
_include at ./loading.jl:2309
include at ./Base.jl:556 [inlined]
```
Since it's not failing on CI, I'm assuming it's microarchitecture specific. My hardware is:
```
processor	: 63
vendor_id	: AuthenticAMD
cpu family	: 25
model		: 17
model name	: AMD EPYC 9354 32-Core Processor
stepping	: 1
microcode	: 0xa10113e
cpu MHz		: 1500.000
cache size	: 1024 KB
```
Bisect says:
```
8db129472ea88a9c5c4a6e94dff7b330723aed28 is the first bad commit
commit 8db129472ea88a9c5c4a6e94dff7b330723aed28
Author: Jameson Nash <vtjnash@gmail.com>
Date:   Sun Feb 4 19:51:02 2024 -0500

    add atomic operators for globals, memory, and setonce (#52868)
```

**Comment by Seelengrab** [comment]
I can reproduce this on an AMD Ryzen 9 7900X.

**Comment by vtjnash** [comment]
Seems to be a regression of https://github.com/JuliaLang/julia/pull/21980
```
julia> let T = Float64, N = 5
           a = ntuple(i -> VecElement(T(i)), N)
           @show Base.datatype_layoutsize(typeof(a))
           @show Base.datatype_alignment(typeof(a))
           @show Base.datatype_layoutsize(Memory{typeof(a)})
           @show Base.datatype_alignment(Memory{typeof(a)})
       end

Base.datatype_layoutsize(typeof(a)) = 64
Base.datatype_alignment(typeof(a)) = 64
Base.datatype_layoutsize(Memory{typeof(a)}) = 64
Base.datatype_alignment(Memory{typeof(a)}) = 64
```

The alignment of 64 exceeds what Julia can provide, so this should not be permitted to construct this vector type, as LLVM will try to over-align it

**Comment by gbaraldi** [comment]
Should we box the vecelements then?

**Comment by vchuravy** [comment]
Proper long-term fix is something along the lines of #21959 and #22649
